src/main/java/ui/components/FilterTextField.java

```diff
@@ -106,6 +117,24 @@ private void setup(Predicate<String> validation) {
         });
     }
 
+    private void handleOnEnter() {
+        if (isNavigating) {
+            suggestion.hide();
+            suggestion.getSelectedContent().ifPresent(this::completeWord);
+        } else {
+            confirmEdit();
+        }
+    }
+
+    private void handleOnEscape() {
+        if (getText().equals(previousText)) {
```

Comment:
Maybe the first escape key press could dismiss the menu, but not revert the field? So in total there would be 3 possible uses for the key.